app/models/propertyDetailsModels.scala

```diff
@@ -85,8 +124,26 @@ object PropertyDetailsRevalued {
   implicit val formats = Json.format[PropertyDetailsRevalued]
 }
 
-case class PropertyDetailsOwnedBefore(isOwnedBefore2012: Option[Boolean] = None,
-                                      ownedBefore2012Value: Option[BigDecimal] = None)
+sealed trait OwnedBeforePolicyYear
+
+case object IsOwnedBefore2012 extends OwnedBeforePolicyYear
+
+case object IsOwnedBefore2017 extends OwnedBeforePolicyYear
```

Comment:
These can be generic as case classes with a start year for a period.

Reply:
I would like to keep them as case objects as they serve as constants and also help in exhaustive matching..

Reply:
yeah that comment was in the general theme of making the 5 year periods automatically and avoiding having to do that. This is fine now.